### ISSUE TITLE
QueryId gets changed when table oid is changed.

### ISSUE THREAD

**Issue by snaga** [uptimejp/sql_firewall#13]
QueryId depends on relation oid instead of relation name, because jumbling query use relation oid.

https://github.com/uptimejp/sql_firewall/blob/develop/sql_firewall.c#L2784

So, queryid would be different in different locations or different timing (c.f. after dump & restore)
even if the table name and query are exact the same ones.

Needs to be fixed to improve portability of the rule file.

**Comment by snaga** [comment]
Fixed at e3b093d20788bfbaccd169d95bc07171d5938459. Not merged yet.

**Comment by snaga** [comment]
Need to have a performance comparison before merging the patch.

**Comment by snaga** [comment]
# jumbling with relid

## pgbench -S -c 8 -t 10000

```
tps = 25589.534904 (including connections establishing)
tps = 25943.138478 (excluding connections establishing)
tps = 25926.877778 (including connections establishing)
tps = 26239.978378 (excluding connections establishing)
tps = 25942.027999 (including connections establishing)
tps = 26500.210014 (excluding connections establishing)
tps = 26853.955101 (including connections establishing)
tps = 27637.641500 (excluding connections establishing)
tps = 27460.843412 (including connections establishing)
tps = 28443.857869 (excluding connections establishing)
```

## pgbench -c 8 -t 1000

```
tps = 268.464871 (including connections establishing)
tps = 269.053856 (excluding connections establishing)
tps = 252.459747 (including connections establishing)
tps = 252.808912 (excluding connections establishing)
tps = 282.829536 (including connections establishing)
tps = 283.454867 (excluding connections establishing)
tps = 242.941789 (including connections establishing)
tps = 243.256689 (excluding connections establishing)
tps = 255.071174 (including connections establishing)
tps = 255.687529 (excluding connections establishing)
```

# jumbling with relname

## pgbench -S -c 8 -t 10000

```
tps = 25972.069636 (including connections establishing)
tps = 26517.057429 (excluding connections establishing)
tps = 25616.730799 (including connections establishing)
tps = 25942.793546 (excluding connections establishing)
tps = 25333.299133 (including connections establishing)
tps = 25882.043879 (excluding connections establishing)
tps = 26798.667972 (including connections establishing)
tps = 27391.397389 (excluding connections establishing)
tps = 27125.172499 (including connections establishing)
tps = 27471.348242 (excluding connections establishing)
```

## pgbench -c 8 -t 1000

```
tps = 271.739675 (including connections establishing)
tps = 272.342011 (excluding connections establishing)
tps = 251.109338 (including connections establishing)
tps = 251.624947 (excluding connections establishing)
tps = 282.011743 (including connections establishing)
tps = 282.652613 (excluding connections establishing)
tps = 252.924559 (including connections establishing)
tps = 253.267972 (excluding connections establishing)
tps = 233.803756 (including connections establishing)
tps = 234.231883 (excluding connections establishing)
```

**Comment by snaga** [comment]
# t-test

## pgbench -S -c 8 -t 10000 (excluding connections establishing)

|  | With relid | With relname |
| --- | --- | --- |
| Mean | 26952.96525 | 26640.9281 |
| Variance | 1106157.515 | 582860.7064 |
| Observations | 5 | 5 |
| Pearson Correlation | 0.84635357 |  |
| Hypothesized Mean Difference | 0 |  |
| df | 4 |  |
| t | 1.214876167 |  |
| P(T<=t) one-tail | 0.145610562 |  |
| t Critical one-tail | 2.131846786 |  |
| P(T<=t) two-tail | 0.291221124 |  |
| t Critical two-tail | 2.776445105 |  |

## pgbench -c 8 -t 1000 (excluding connections establishing)

|  | With relid | With relname |
| --- | --- | --- |
| Mean | 254.7231796 | 258.8238852 |
| Variance | 86.27687585 | 359.5018622 |
| Observations | 5 | 5 |
| Pearson Correlation | 0.286575388 |  |
| Hypothesized Mean Difference | 0 |  |
| df | 4 |  |
| t | -0.493782918 |  |
| P(T<=t) one-tail | 0.323672165 |  |
| t Critical one-tail | 2.131846786 |  |
| P(T<=t) two-tail | 0.647344329 |  |
| t Critical two-tail | 2.776445105 |  |

**Comment by snaga** [comment]
The performance penalty is not significant.
Merged 23317f1c129b24e9af841cd17a10bee54cc33498.
